src/Atlas.cc

```diff
@@ -259,6 +259,81 @@ bool Atlas::isImuInitialized()
     return mpCurrentMap->isImuInitialized();
 }
 
+void Atlas::PreSave()
+{
+    if(mpCurrentMap){
+        if(!mspMaps.empty() && mnLastInitKFidMap < mpCurrentMap->GetMaxKFid())
+            mnLastInitKFidMap = mpCurrentMap->GetMaxKFid()+1; //The init KF is the next of current maximum
+    }
+
+    struct compFunctor
+    {
+        inline bool operator()(Map* elem1 ,Map* elem2)
+        {
+            return elem1->GetId() < elem2->GetId();
+        }
+    };
+    std::copy(mspMaps.begin(), mspMaps.end(), std::back_inserter(mvpBackupMaps));
+    sort(mvpBackupMaps.begin(), mvpBackupMaps.end(), compFunctor());
+
+    std::set<GeometricCamera*> spCams(mvpCameras.begin(), mvpCameras.end());
+    cout << "There are " << spCams.size() << " cameras in the atlas" << endl;
+    for(Map* pMi : mvpBackupMaps)
+    {
+        cout << "Pre-save of map " << pMi->GetId() << endl;
+        pMi->PreSave(spCams);
+    }
+    cout << "Maps stored" << endl;
+    for(GeometricCamera* pCam : mvpCameras)
+    {
+        cout << "Pre-save of camera " << pCam->GetId() << endl;
+        if(pCam->GetType() == pCam->CAM_PINHOLE)
+        {
+            mvpBackupCamPin.push_back((Pinhole*) pCam);
+        }
+        else if(pCam->GetType() == pCam->CAM_FISHEYE)
+        {
+            mvpBackupCamKan.push_back((KannalaBrandt8*) pCam);
+        }
+    }
+
+}
+
+void Atlas::PostLoad()
+{
+    mvpCameras.clear();
+    map<unsigned int,GeometricCamera*> mpCams;
+    for(Pinhole* pCam : mvpBackupCamPin)
+    {
+        //mvpCameras.push_back((GeometricCamera*)pCam);
+        mvpCameras.push_back(pCam);
+        mpCams[pCam->GetId()] = pCam;
+    }
+    for(KannalaBrandt8* pCam : mvpBackupCamKan)
+    {
+        //mvpCameras.push_back((GeometricCamera*)pCam);
+        mvpCameras.push_back(pCam);
+        mpCams[pCam->GetId()] = pCam;
+    }
+
+    mspMaps.clear();
+    unsigned long int numKF = 0, numMP = 0;
+    map<long unsigned int, KeyFrame*> mpAllKeyFrameId;
+    for(Map* pMi : mvpBackupMaps)
```

Comment:
I use for(Map* pMi : mspMaps) here directly, since mspMaps is already stored and mvpBackupMaps may be not necessary